### PR TITLE
OTT-1262: Fix panic observed for hybrid endpoint when site object is nil

### PR DESCRIPTION
**Pull request by ashishshinde-pubm**
@ashishshinde-pubm : Please check my review comments.
1. I could see similar issue in https://github.com/prebid/prebid-server/blob/961bb8357202142b9b5f4ab6ebe5c71bdd20a356/endpoints/openrtb2/auction.go#L2006  - Let us see if we can fix this to in prebid-pr / in OW if we are using that function. Even that can cause panic

### CODE REVIEW

File: endpoints/openrtb2/auction.go [hunk 1]
```diff
@@ -1999,12 +1999,12 @@ func setSiteImplicitly(httpReq *http.Request, r *openrtb_ext.RequestWrapper) {
 	}
 }
 
-func setSitePageIfEmpty(site *openrtb2.Site, sitePage string) {
-	if site == nil {
-		site = &openrtb2.Site{}
+func setSitePageIfEmpty(r *openrtb_ext.RequestWrapper, sitePage string) {
```
Comment (ShriprasadM) [inline endpoints/openrtb2/auction.go:2002]: Why there is a need of passing entire `*openrtb_ext.RequestWrapper,` ? Can we keep accepting `*openrtb2.Site`?

**Review by ShriprasadM — CHANGES_REQUESTED** [review]
Please check review comments

File: endpoints/openrtb2/auction.go [hunk 2]
```diff
@@ -1983,6 +1983,9 @@ func setSiteImplicitly(httpReq *http.Request, r *openrtb_ext.RequestWrapper) {
 	}
 
 	if referrerCandidate != "" {
+		if r.Site == nil {
```
Comment (pm-isha-bharti) [inline endpoints/openrtb2/auction.go:1986]: We are already doing this inside the function `setSitePageIfEmpty`, what is the point of moving it outside the function, are we solving something here?

**Review by ShriprasadM — APPROVED** [review]
LGTM

**Review by pm-jaydeep-mohite — APPROVED** [review]
LGTM

**Pull request merged**